### PR TITLE
AIAT-204 Add optional kort db user and url connection options

### PR DESCRIPTION
**Pull request by sconway-scottlogic**
- Added optional `MONGO_INITDB_ROOT_USERNAME` and `MONGO_INITDB_ROOT_PASSWORD` environment variables to define credentials for a root Mongodb user. This uses existing Mongodb functionality.
-  Added optional `MONGO_KORT_USERNAME` and `MONGO_KORT_PASSWORD` environment variables to define credentials for a  Mongodb Kort DB user. This user is created by the new `mongo-init.js` script.
- If a `MONGO_OPTIONS`  environment variable is defined then it is added to the connection string used by Kort.
- Updated README

See [AIAT-204 Update MongoDB connection url to support credentials and list of options](https://scottlogic.atlassian.net/browse/AIAT-204)

Tested manually in Docker:

- With no credentials defined. Kort works as expected.
- With credentials defined. Kort works as expected and the Mongo Kort DB username can be seen in the Mongo logs.
- With `MONGO_KORT_USERNAME` defined but not `MONGO_KORT_PASSWORD`. An error is logged in the `mongo` and `kort` containers.
- With `MONGO_KORT_PASSWORD` defined but not `MONGO_KORT_USERNAME`. An error is logged in the `mongo` and `kort` containers.
- With credentials and `MONGO_OPTIONS` defined as `"connectTimeoutMS=1"`. Network timeouts are seen in the `kort` log.
- With credentials and `MONGO_OPTIONS` defined as `"connectTimeoutMS=10000"`. Kort works as expected, without timeouts in the `kort` log.

Note that when changing credentials, the containers should be deleted and the `data` directory deleted, so that the `mongo-init.js` script is run at start-up. The images need to be rebuilt.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
+mongoURL += mongoHost + ":" + MONGO_PORT;
+mongoURL += "/" + MONGO_DATABASE;
+if (process.env.MONGO_OPTIONS) {
+  mongoURL += "?" + process.env.MONGO_OPTIONS;
 }
 
+const port = process.env.PORT || 3000;
+
 //load in environment variables
 require('dotenv').config();
```
Comment (pthanvi) [inline app.js]: This should probably be moved to the top before we use any environment variables

**Comment by pthanvi** [comment]
I know this code is not touched by you, but can you change the user name and password for kort application from `DEFAULT_ADMIN_USER` and `DEFAULT_ADMIN_PASSWORD` to something more meaningful (line 38) - we would also need to change env file and README for this.

File: app.js (hunk 1 above)
Comment (sconway-scottlogic) [reply app.js]: Done

**Comment by sconway-scottlogic** [comment]
> I know this code is not touched by you, but can you change the user name and password for kort application from `DEFAULT_ADMIN_USER` and `DEFAULT_ADMIN_PASSWORD` to something more meaningful (line 38) - we would also need to change env file and README for this.

I have changed the variable names to `kortAdminUser` and `kortAdminPassword`, the default value constant names to `DEFAULT_KORT_ADMIN_USER` and `DEFAULT_KORT_ADMIN_PASSWORD` and the environment variables to `KORT_ADMIN_USER` and `KORT_ADMIN_PASSWORD`. I have updated the README.

**Review by pthanvi — APPROVED** [review]

**Review by kclark-scottlogic — APPROVED** [review]
Tester Approved

**Pull request merged**